src/main/java/com/ning/compress/lzf/impl/UnsafeChunkEncoderBE.java

```diff
@@ -85,6 +85,7 @@ protected int tryCompress(byte[] in, int inPos, int inEnd, byte[] out, int outPo
             hashTable[hash(seen)] = inPos;
             ++inPos;
         }
+        assert inPos <= inEnd + TAIL_LENGTH;
```

Comment:
No `assert`s please: if we want to check it, check and throw `IllegalStateException`; if not, just remove.